File: nestjs_drizzle/src/common/api-exception.ts
/**
 * 비즈니스 예외 (SR §3, §5). resultCode 를 담아 던지며,
 * 전역 예외 필터가 이를 표준 응답(HTTP 200)으로 변환한다.
 */
export class ApiException extends Error {
  constructor(readonly resultCode: number) {
    super(`ApiException(resultCode=${resultCode})`);
    this.name = 'ApiException';
  }
}

File: spring_webflux_jooq/src/main/java/me/yeonhyuk/heavy_traffic/spring_webflux_jooq/common/ApiException.java
package me.yeonhyuk.heavy_traffic.spring_webflux_jooq.common;

import lombok.Getter;

/**
 * 비즈니스 로직/인증 수행 중 발생하는, resultCode 로 변환 가능한 예외.
 */
@Getter
public class ApiException extends RuntimeException {

    private final ResultCodeEnum resultCode;

    public ApiException(ResultCodeEnum resultCode) {
        super(resultCode.name());
        this.resultCode = resultCode;
    }

    public ApiException(ResultCodeEnum resultCode, Throwable cause) {
        super(resultCode.name(), cause);
        this.resultCode = resultCode;
    }
}
